src/domain/bitcoin/onchain/index.types.d.ts

```diff
@@ -65,3 +71,63 @@ interface IOnChainService {
     targetConfirmations: TargetConfirmations,
   ): Promise<Satoshis | OnChainServiceError>
 }
+
+interface IBitcoindService {
+  getBlockCount(): Promise<number>
+  getBlockchainInfo(): Promise<{ chain: BtcNetwork }>
+  createWallet(
+    walletName: BitcoindWalletName,
+  ): Promise<
+    | { name: BitcoindWalletName; warning: string }
+    | UnknownRepositoryError
+    | WalletAlreadyExistError
+  >
+  listWallets(): Promise<BitcoindWalletName[]>
+  loadWallet(walletName: BitcoindWalletName): Promise<{ name: string; warning: string }>
+  unloadWallet(walletName: BitcoindWalletName): Promise<{ warning: string }>
+}
+
+interface IBitcoindWalletService {
```

Comment:
I don't see the reason for having `bitcoind` in the name. This could also be deconstructed into an `AddressRepository` a `PsbtProcessor` and a `TransactionBroadcaster`.

Having smaller interfaces is valuable because it reduces the coupling between use cases.